src/status_im2/contexts/wallet/common/utils.cljs

```diff
@@ -33,3 +23,24 @@
   [number-of-accounts]
   (let [path (get-derivation-path number-of-accounts)]
     (format-derivation-path path)))
+
+(defn- calculate-raw-balance
+  [raw-balance decimals]
+  (if-let [n (utils.number/parse-int raw-balance nil)]
+    (/ n (Math/pow 10 (utils.number/parse-int decimals)))
+    0))
+
+(defn- total-token-value-in-all-chains
+  [{:keys [balances-per-chain decimals]}]
+  (->> balances-per-chain
+       (vals)
+       (map #(calculate-raw-balance (:raw-balance %) decimals))
+       (reduce +)))
+
+(defn calculate-balance
```

Comment:
Would be great to add unit tests for this function (also for every utils public function, but that can be done on a separate PR), wdyt?

Reply:
+1 on this! Although I think we should do in a follow up - would be great to have this pr merged in asap as it will likely affect other prs but the changes brought in here are the right way to go 🚀

Reply:
Yes, I'm trying to merge it ASAP.

And bout that function, it already existed in the code and it's being used by a subscription (`:wallet/balances`), that sub already has tests.

I refactored the implementation but it's still passing the tests, so it's already being tested in some way.